src/controllers/product.js

```diff
@@ -235,11 +235,48 @@ const orderRequest = async (req, res) => {
   }
 };
 
+const getRequesters = async (req, res) => {
+  try {
+    const product = await ProductModel.findById(req.params.productId);
+    if (!product) {
+      return res.status(404).json({
+        message: 'Product not found',
+      });
+    }
+
+    if (String(product.publisher) !== req.user.userId) {
+      return res.status(401).json({
+        message: 'You are not authorized to perform this action',
+      });
+    }
+
+    if (product.postType !== 'Donate') {
```

Comment:
I don't think we need this

Reply:
Actually there were places where when I was testing, I did get this error, because I was trying to see `requesters` of a "requested" product. FE will not be using it, but it doesn't affect the app. I still can remove it upon your request. Just let me know.